### ISSUE TITLE
deis registry:set fails to create private-registry secret

### ISSUE THREAD

**Issue by seanknox** [deis/controller#1031]
The first time I ran `deis registry:set`:
```
$ deis registry:set username=myusername password=mypassword   -a apple
Applying registry information... Error: Unknown Error (400): {"detail":"(app::deploy): There was a problem while deploying v7 of apple-worker. Going back to the previous release. Additional information:
('failed to failed to create Secret \"private-registry\" in Namespace \"apple\": 409 Conflict', 'private-registry', 'apple')"}
```

Ran the same command once more and it worked. Full output from the controller:

```
ERROR [perftest]: (app::deploy): There was a problem while deploying v7 of perftest-worker. Going back to the previous release. Additional information:
('failed to failed to create Secret "private-registry" in Namespace "perftest": 409 Conflict', 'private-registry', 'perftest')
ERROR:root:(app::deploy): There was a problem while deploying v7 of perftest-worker. Going back to the previous release. Additional information:
('failed to failed to create Secret "private-registry" in Namespace "perftest": 409 Conflict', 'private-registry', 'perftest')
Traceback (most recent call last):
  File "/app/scheduler/__init__.py", line 528, in _set_image_secret
    self.get_secret(namespace, secret_name)
  File "/app/scheduler/__init__.py", line 1032, in get_secret
    'get Secret "{}" in Namespace "{}"', name, namespace
scheduler.KubeHTTPException: ('failed to get Secret "private-registry" in Namespace "perftest": 404 Not Found', 'private-registry', 'perftest')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/scheduler/__init__.py", line 126, in deploy
    namespace, name, image, entrypoint, command, **kwargs
  File "/app/scheduler/__init__.py", line 1573, in update_deployment
    **kwargs)
  File "/app/scheduler/__init__.py", line 1803, in _build_deployment_manifest
    manifest['spec']['template'] = self._build_pod_manifest(namespace, name, image, **kwargs)
  File "/app/scheduler/__init__.py", line 291, in _build_pod_manifest
    self._set_image_secret(spec, namespace, **kwargs)
  File "/app/scheduler/__init__.py", line 534, in _set_image_secret
    secret_type='kubernetes.io/dockerconfigjson'
  File "/app/scheduler/__init__.py", line 1099, in create_secret
    'failed to create Secret "{}" in Namespace "{}"', name, namespace
scheduler.KubeHTTPException: ('failed to failed to create Secret "private-registry" in Namespace "perftest": 409 Conflict', 'private-registry', 'perftest')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/api/models/app.py", line 551, in deploy
    async_run(tasks)
  File "/app/api/utils.py", line 169, in async_run
    raise error
  File "/usr/lib/python3.5/asyncio/tasks.py", line 241, in _step
    result = coro.throw(exc)
  File "/app/api/utils.py", line 182, in async_task
    yield from loop.run_in_executor(None, params)
  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/app/scheduler/__init__.py", line 139, in deploy
    ) from e
scheduler.KubeException: There was a problem while deploying v7 of perftest-worker. Going back to the previous release. Additional information:
('failed to failed to create Secret "private-registry" in Namespace "perftest": 409 Conflict', 'private-registry', 'perftest')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/api/views.py", line 273, in post_save
    config.app.deploy(self.release)
  File "/app/api/models/app.py", line 555, in deploy
    raise ServiceUnavailable(err) from e
api.exceptions.ServiceUnavailable: (app::deploy): There was a problem while deploying v7 of perftest-worker. Going back to the previous release. Additional information:
('failed to failed to create Secret "private-registry" in Namespace "perftest": 409 Conflict', 'private-registry', 'perftest')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/rest_framework/views.py", line 471, in dispatch
    response = handler(request, *args, **kwargs)
  File "/app/api/views.py", line 181, in create
    return super(AppResourceViewSet, self).create(request, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/rest_framework/mixins.py", line 21, in create
    self.perform_create(serializer)
  File "/app/api/viewsets.py", line 21, in perform_create
    self.post_save(obj)
  File "/app/api/views.py", line 276, in post_save
    raise DeisException(str(e)) from e
api.exceptions.DeisException: (app::deploy): There was a problem while deploying v7 of perftest-worker. Going back to the previous release. Additional information:
('failed to failed to create Secret "private-registry" in Namespace "perftest": 409 Conflict', 'private-registry', 'perftest')
```

**Comment by seanknox** [comment]
@helgi what's the best way to apply the fix today? Clone the repo and build a container image?

**Comment by helgi** [comment]
For a merged commit you can do either that or go to https://quay.io/repository/deis/controller?tag=latest&tab=tags and find the commit in the PR and correlate... not sure why that hasn't happened in this case just yet.

It's available in https://quay.io/repository/deisci/controller?tag=latest&tab=tags
> helgi merged commit 1ca970b into deis:master an hour ago

quay.io/deisci/controller:git-1ca970b
